### PR TITLE
review query_learn

### PR DESCRIPTION
**Pull request by bkj**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -126,53 +112,61 @@ def main(args):
         model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[args.gpu])
         model_without_ddp = model.module
     n_parameters = sum(p.numel() for p in model.parameters() if p.requires_grad)
+    params = [p for n, p in model_without_ddp.named_parameters() if "backbone" not in n and p.requires_grad]
```
Comment (bkj) [inline main.py]: @ezekielbarnett this might be wrong -- in the original they set a different learning rate for `"backbone" not in n` and `"backbone" in n`.  Your version sets the same LR for both

File: models/detr.py [hunk 2]
```diff
@@ -58,11 +58,20 @@ def forward(self, samples: NestedTensor):
...
         if isinstance(samples, (list, torch.Tensor)):
             samples = nested_tensor_from_tensor_list(samples)
-        features, pos = self.backbone(samples)
+
+        # embed image
+        features, pos        = self.backbone(samples)
+
+        # embed query
+        query_fts, query_pos = self.backbone(query)
+
```
Comment (bkj) [inline models/detr.py:86]: Don't you have to pool the query somehow?  I didn't look closely at their `backbone`, but I think it returns the full feature map, not the global average pooled version which I'd think we want to use

File: models/detr.py [hunk 3]
```diff
@@ -58,11 +58,20 @@ def forward(self, samples: NestedTensor):
...
+        # expend query to num_queries
+        query_fts = torch.cat(self.num_queries*[query_fts])
 
         src, mask = features[-1].decompose()
         assert mask is not None
-        hs = self.transformer(self.input_proj(src), mask, self.query_embed.weight, pos[-1])[0]
+        #hs = self.transformer(self.input_proj(src), mask, self.query_embed.weight, pos[-1])[0]
+        hs = self.transformer(self.input_proj(src), mask, query_fts, pos[-1])[0]
```
Comment (bkj) [inline models/detr.py]: Looking at this it probably makes more sense to add `query_fts` to `self.query_embed.weight` instead of just repeating `query_fts`.  I don't 100% understand what `query_embed` learns, but I think that's more reasonable.
